RecoTracker/LSTCore/src/alpaka/LSTESData.dev.cc

```diff
@@ -0,0 +1,114 @@
+                    std::shared_ptr<SDL::ModuleConnectionMap<SDL::Dev>> moduleConnectionMap) {
+    // Module orientation information (DrDz or phi angles)
+    auto endcap_geom =
+        get_absolute_path_after_check_file_exists(trackLooperDir() + "/data/OT800_IT615_pt0.8/endcap_orientation.bin");
+    auto tilted_geom = get_absolute_path_after_check_file_exists(
+        trackLooperDir() + "/data/OT800_IT615_pt0.8/tilted_barrel_orientation.bin");
+    // Module connection map (for line segment building)
+    auto mappath = get_absolute_path_after_check_file_exists(
+        trackLooperDir() + "/data/OT800_IT615_pt0.8/module_connection_tracing_merged.bin");
+
+    endcapGeometry->load(endcap_geom);
+    tiltedGeometry->load(tilted_geom);
+    moduleConnectionMap->load(mappath);
+
+    auto pLSMapDir = trackLooperDir() + "/data/OT800_IT615_pt0.8/pixelmap/pLS_map";
+    const std::array<std::string, 4> connects{
+        {"_layer1_subdet5", "_layer2_subdet5", "_layer1_subdet4", "_layer2_subdet4"}};
+    std::string path;
+
+    static_assert(connects.size() == std::tuple_size<std::decay_t<decltype(pLStoLayer[0])>>{});
+    for (unsigned int i = 0; i < connects.size(); i++) {
+      auto connectData = connects[i].data();
+
+      path = pLSMapDir + connectData + ".bin";
+      pLStoLayer[0][i] = SDL::ModuleConnectionMap<SDL::Dev>(get_absolute_path_after_check_file_exists(path));
+
+      path = pLSMapDir + "_pos" + connectData + ".bin";
+      pLStoLayer[1][i] = SDL::ModuleConnectionMap<SDL::Dev>(get_absolute_path_after_check_file_exists(path));
+
+      path = pLSMapDir + "_neg" + connectData + ".bin";
+      pLStoLayer[2][i] = SDL::ModuleConnectionMap<SDL::Dev>(get_absolute_path_after_check_file_exists(path));
+    }
+  }
+}  // namespace
+
+std::unique_ptr<SDL::LSTESData<SDL::DevHost>> SDL::loadAndFillESHost() {
+  uint16_t nModules;
+  uint16_t nLowerModules;
+  unsigned int nPixels;
+  std::shared_ptr<SDL::modulesBuffer<SDL::DevHost>> modulesBuffers = nullptr;
+  auto pLStoLayer = std::make_shared<SDL::MapPLStoLayer>();
+  auto endcapGeometry = std::make_shared<SDL::EndcapGeometry<SDL::Dev>>();
+  auto tiltedGeometry = std::make_shared<SDL::TiltedGeometry<SDL::Dev>>();
+  auto pixelMapping = std::make_shared<SDL::pixelMap>();
+  auto moduleConnectionMap = std::make_shared<SDL::ModuleConnectionMap<SDL::Dev>>();
+  ::loadMapsHost(*pLStoLayer, endcapGeometry, tiltedGeometry, moduleConnectionMap);
+
+  auto endcapGeometryBuffers =
+      std::make_shared<SDL::EndcapGeometryBuffer<SDL::DevHost>>(cms::alpakatools::host(), endcapGeometry->nEndCapMap);
+  alpaka::QueueCpuBlocking queue(cms::alpakatools::host());
+  alpaka::memcpy(
+      queue, endcapGeometryBuffers->geoMapDetId_buf, endcapGeometry->geoMapDetId_buf, endcapGeometry->nEndCapMap);
+  alpaka::memcpy(
+      queue, endcapGeometryBuffers->geoMapPhi_buf, endcapGeometry->geoMapPhi_buf, endcapGeometry->nEndCapMap);
```

Comment:
I'm confused what happens here. First `SDL::EndcapGeometry<SDL::Dev>` is created, and used to read some data from file(s). The type pretends to depend on the Device type, but really doesn't (which I commented earlier). Then data from the `SDL::EndcapGeometry<SDL::Dev>` is copied to `SDL::EndcapGeometryBuffer<SDL::DevHost>`.

Couldn't these `memcpy()` calls use just the `std::memcpy()` since everything is on the host?

Or, event better, could `loadsMapsHost` read the data directly to `SDL::EndcapGeometryBuffer<SDL::DevHost>`?